it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -943,124 +943,208 @@ class PersistenceConnectorSpec
+      }
     }
-    "On an incorrect Json fragment, must return a JsResult.Exception" in forAll(arbitrary[MovementId], messageType) {
-      (departureId, messageType) =>
-        server.stubFor(
-          post(
-            urlEqualTo(targetUrl(departureId))
+
+    "when posting without body and message type (creating an empty message)" - {
+      "On successful update of an element, must return ACCEPTED" in forAll(arbitrary[MovementId], okResultGen) {
+        (departureId, resultRes) =>
+          server.stubFor(
+            post(
+              urlEqualTo(targetUrl(departureId))
+            )
+              .willReturn(
+                aResponse().withStatus(OK).withBody(Json.stringify(Json.toJson(resultRes)))
+              )
           )
-            .withHeader(HeaderNames.CONTENT_TYPE, equalTo(MimeTypes.XML))
-            .withHeader("X-Message-Type", equalTo(messageType.code))
-            .willReturn(
-              aResponse()
-                .withStatus(OK)
-                .withBody(
-                  "{ hello"
-                )
+
+          implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = Seq(HeaderNames.ACCEPT -> ContentTypes.JSON))
+
+          whenReady(persistenceConnector.postMessage(departureId, None, None)) {
+            result =>
+              result mustBe resultRes
+          }
+      }
+
+      "On an upstream internal server error, get a UpstreamErrorResponse" in forAll(arbitrary[MovementId]) {
+        departureId =>
+          server.stubFor(
+            post(
+              urlEqualTo(targetUrl(departureId))
             )
-        )
+              .willReturn(
+                aResponse()
+                  .withStatus(INTERNAL_SERVER_ERROR)
+                  .withBody(
+                    Json.stringify(Json.toJson(PresentationError.internalServiceError()))
+                  )
+              )
+          )
 
-        implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = Seq(HeaderNames.ACCEPT -> ContentTypes.JSON))
+          implicit val hc: HeaderCarrier = HeaderCarrier(extraHeaders = Seq(HeaderNames.ACCEPT -> ContentTypes.JSON))
 
-        val source = Source.single(ByteString(<test></test>.mkString, StandardCharsets.UTF_8))
+          val future = persistenceConnector.postMessage(departureId, None, None).map(Right(_)).recover {
+            case NonFatal(e) => Left(e)
+          }
 
-        val future = persistenceConnector.postMessage(departureId, messageType, source).map(Right(_)).recover {
-          case NonFatal(e) => Left(e)
-        }
+          whenReady(future) {
+            result =>
+              result.left.toOption.get mustBe a[UpstreamErrorResponse]
+              val response = result.left.toOption.get.asInstanceOf[UpstreamErrorResponse]
+              response.statusCode mustBe INTERNAL_SERVER_ERROR
+              Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Internal server error", ErrorCode.InternalServerError))
```

Comment:
Not inherently wrong (though I think this throws warnings in Scala 2.13), but you may prefer something like:

```scala
whenReady(future) {
    case Left(UpstreamErrorResponse(`message`, INTERNAL_SERVER_ERROR, _,, _)) =>
              Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Internal server error", ErrorCode.InternalServerError))
    case other => fail(s"Expected internal server error, got $other")
}
```

I don't see this as a blocker though.